### ISSUE TITLE
Multiple typescript projects in the same web site

### ISSUE THREAD

**Issue by tsachyg** [Microsoft/TypeScript#1828]
I am working on CDN web site and there is many projects and many versions of those projects...
I had to set the build action of all "ts" files to Content instead of "TypeScriptCompile" otherwise the web site build operation fails.
It would be nice if I could setup each version of a project as typescript project and let it compile when building the whole CDN web site.

**Comment by mhegazy** [comment]
I will need more details on your setup to be able to help you better. but in general. you should have a 1-to-1 mapping between your projects and your outputs. think of a project as one call to tsc.exe on the commandline, if you have multiple calls, i.e. building same set of files with different configurations, then you would need to have a different project for you.I would need some more details though to be able to help out.

**Comment by tsachyg** [comment]
Let's say my web site host name is cdn.local.com so a project should be located in http://can.local.com/lib/<ProjectName>/<Version>/There is also controller in this web site to translate the URLs to the closest version (e.g. Latest brings the latest project version and 3 brings the latest minor version of the major version 3 and so on)Each project can have many versions (minor or major) and they can not be compiled together.Each project can have references to other projects (so we need to be careful with circular references)
I think that each version of project should have a project file (tsproj like csproj for c# project) with a list of all the relevant ts files in the project and list of references to another project files (just like dll references).

Sent from Outlook

    _____________________________

**Comment by mhegazy** [comment]
> I think that each version of project should have a project file (tsproj like csproj for c# project) with a list of all the relevant ts files in the project and list of references to another project files (just like dll references).

This should work today. TypeScript does not have a special project file, we just use any project file. So just create a .csproj with all your .ts files in this project, and have one for every project.

**Comment by tsachyg** [comment]
It may work. I will try it during this week. But, what about references to other projects? Does tsc.exe take care of project references?

Sent from Outlook




On Fri, Jan 30, 2015 at 4:10 PM -0800, "Mohamed Hegazy" <notifications@github.com> wrote:













I think that each version of project should have a project file (tsproj like csproj for c# project) with a list of all the relevant ts files in the project and list of references to another project files (just like dll references).




This should work today. TypeScript does not have a special project file, we just use any project file. So just create a .csproj with all your .ts files in this project, and have one for every project.

**Comment by mhegazy** [comment]
For references it depends on how you package them. if i understand correctly, each project is a standalone application, they just share some sources. if this is correct, then the easiest way is to use ///  <references> to shared sources in each project.

**Comment by tsachyg** [comment]
OKI created a project file for one of the projects in the site and it seems to work, though, I had to modify the csproj file manually in order to add the ts files from the same location as in the web site and I had to include typescript compilation instruction inside the csproj file (otherwise ts files will not compile). /// references works. So it seems like my problem solved, though, this whole solution looks to me like a workaround and there should be a better way to do it. My reasons are 1. because of having to modify the csproj file manually2. because /// references looks like a patch.
Anyway, it does work and for now I think this is good enough.

Sent from Outlook

    _____________________________
